ngraph/core/src/op/if.cpp

```diff
@@ -0,0 +1,292 @@
+        {
+            auto description_find_lambda = [=](MultiSubgraphOutputDescriptionPtr& descr) {
+                return descr->m_output_index == static_cast<uint64_t>(output_index);
+            };
+
+            auto then_output_description = *find_if(m_output_descriptions[then_body_index].begin(),
+                                                    m_output_descriptions[then_body_index].end(),
+                                                    description_find_lambda);
+            auto else_output_description = *find_if(m_output_descriptions[else_body_index].begin(),
+                                                    m_output_descriptions[else_body_index].end(),
+                                                    description_find_lambda);
+            auto then_out_node = m_bodies[then_body_index]
+                                     ->get_results()
+                                     .at(then_output_description->m_body_value_index)
+                                     ->input_value(0);
+            auto else_out_node = m_bodies[else_body_index]
+                                     ->get_results()
+                                     .at(else_output_description->m_body_value_index)
+                                     ->input_value(0);
+            auto then_node_partial_shape = then_out_node.get_partial_shape();
+            auto else_node_partial_shape = else_out_node.get_partial_shape();
+            NGRAPH_CHECK(then_out_node.get_element_type() == else_out_node.get_element_type(),
+                         "type of then_body output is not equal type of else_body output");
+
+            auto partial_shape = resolve_shape(then_node_partial_shape, else_node_partial_shape);
+            set_output_type(output_index, then_out_node.get_element_type(), partial_shape);
+        }
+    }
+}
+
+void op::v8::If::fill_body(const std::shared_ptr<op::v8::If>& new_op,
+                           size_t branch_index,
+                           const OutputVector& new_args) const
+{
+    auto body = m_bodies[branch_index];
+    auto param_size = body->get_parameters().size();
+    std::vector<::ngraph::element::Type> types(param_size);
+    std::vector<::ngraph::PartialShape> new_shapes(param_size);
+    auto& input_descriptions = m_input_descriptions[branch_index];
+    size_t parameters_num = 0;
+    for (auto& input_description : input_descriptions)
+    {
+        if (input_description->m_input_index < new_args.size())
+        {
+            types[input_description->m_body_parameter_index] =
+                new_args[input_description->m_input_index].get_element_type();
+            new_shapes[input_description->m_body_parameter_index] =
+                new_args[input_description->m_input_index].get_partial_shape();
+            parameters_num++;
+        }
+    }
+    auto func =
+        std::make_shared<Function>(body->get_results(), body->get_sinks(), body->get_parameters());
+    auto spec_func =
```

Comment:
why do we use `specialize_function `here? please use `clone_function`
Please compare your approach with Loop op